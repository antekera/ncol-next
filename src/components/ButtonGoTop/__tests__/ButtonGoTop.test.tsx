import { render, screen, fireEvent } from '@testing-library/react'

import { ButtonGoTop } from '..'

const spyScrollTo = jest.fn()

describe('ButtonGoTop', () => {
  beforeEach(() => {
    Object.defineProperty(global.window, 'scrollTo', { value: spyScrollTo })
    spyScrollTo.mockClear()
  })
  test('should match snapshot', () => {
    const { container } = render(<ButtonGoTop />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should call window scroll onClick', () => {
    render(<ButtonGoTop />)
    const button = screen.getByTestId('button-go-top')
    fireEvent.click(button)
    expect(spyScrollTo).toHaveBeenCalledWith({
      left: 0,
      top: 0,
      behavior: 'smooth'
    })
  })
})
