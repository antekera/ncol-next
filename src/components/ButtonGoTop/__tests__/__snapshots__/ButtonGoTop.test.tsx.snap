// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGoTop should match snapshot 1`] = `
<button
  class="link-go-top absolute -top-3 right-6 h-9 w-9 cursor-pointer rounded border-none bg-primary text-white duration-150 ease-in hover:-top-4"
  data-testid="button-go-top"
>
  <svg
    class="lucide lucide-chevron-up"
    fill="none"
    height="36"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m18 15-6-6-6 6"
    />
  </svg>
</button>
`;
